hammerblade/torch/kernel/kernel_cat.cpp

```diff
@@ -0,0 +1,44 @@
+//====================================================================
+// _cat kernel
+//====================================================================
+// simple _cat kernel only works with 0 dim
+//
+// Authors : Lin Cheng, Janice Wei
+// Date    : 07/29/2020, 08/04/2020
+
+
+#include <kernel_common.hpp>
+
+extern "C" {
+
+//====================================================================
+// tensorlib__cat
+//====================================================================
+// This is a simple _cat kernel only works with 0 dim
+
+__attribute__ ((noinline))
+int tensorlib__cat( hb_tensor_t** tensors_p, hb_tensor_t* result_p,
+                    uint32_t* length_p, int32_t* dim_p)
+{
+  HBTensor<float> result(result_p);
+  uint32_t length = *length_p;
+  int32_t dim = *dim_p;
+  size_t index = 0;
+
+  bsg_cuda_print_stat_kernel_start();
+  for (size_t i = 0; i < length; i++) {
+	HBTensor<float> tensor(tensors_p[i]);
+	for (size_t j = 0; j < tensor.numel(); j++) {
+	  result(index) = tensor(j);
+	  index = index + 1;
+	}
+  }
+
```

Comment:
Hmmm, this is a little scary -- basically you have every tile do the same work. If you are not ready to parallelize the work, you should put `if (__bsg_id == 0)`, so tile 0 will do the work while others staying idle